### PR TITLE
Add unit tests for BaseRegisterFragment class

### PR DESCRIPTION
**Pull request by ndegwamartin**

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5432800&bid=17815738)

File: opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    private ArgumentCaptor<String> openSRPIdArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Before
+    public void setUp() {
+
+        MockitoAnnotations.initMocks(this);
+        baseRegisterFragment = Mockito.mock(BaseRegisterFragment.class, Mockito.CALLS_REAL_METHODS);
```
Comment (ekigamba) [inline opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java:84]: Nice one :+1:

File: opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+        Mockito.doReturn(opensrpContext).when(baseRegisterFragment).context();
+        Mockito.doReturn(RuntimeEnvironment.application.getResources().getString(R.string.search_hint)).when(opensrpContext).getStringResource(R.string.search_hint);
+
+        String hint = RuntimeEnvironment.application.getResources().getString(R.string.search_hint);
+        Assert.assertEquals(hint, provider.searchHint());
+    }
+
+    @Test
+    @Ignore
```
Comment (ekigamba) [inline opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java:115]: Create an issue to fix this test & track it

File: opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+        baseRegisterFragment = Mockito.spy(baseRegisterFragment);
+
+        Mockito.doReturn(searchCancelView).when(baseRegisterFragment).getSearchCancelView();
+
+        Mockito.doNothing().when(baseRegisterFragment).filter(ArgumentMatchers.eq(OPENSRP_ID_NO_HYPHENS), ArgumentMatchers.eq(""), ArgumentMatchers.anyString(), ArgumentMatchers.eq(true));
+
+        baseRegisterFragment.onQRCodeSucessfullyScanned(OPENSRP_ID);
+
+        Mockito.verify(baseRegisterFragment).filter(openSRPIdArgumentCaptor.capture(), stringArgumentCaptor.capture(), stringArgumentCaptor.capture(), qrCodeArgumentCaptor.capture());
```
Comment (ekigamba) [inline opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java:180]: Verify the boolean passed in the `filter` method as the last param

File: opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java (hunk 2 above)
Comment (ndegwamartin) [reply opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java:115]: Done, [Issue 509](https://github.com/OpenSRP/opensrp-client-core/issues/509)

File: opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java (hunk 3 above)
Comment (ndegwamartin) [reply opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java:180]: Done

**Pull request merged**